pallets/payment-streams/src/utils.rs

```diff
@@ -202,25 +196,29 @@ where
             });
         }
 
-        // Update the user's deposit based on the new rate
-        let new_deposit = new_rate
-            .checked_mul(&T::BlockNumberToBalance::convert(T::NewStreamDeposit::get()))
-            .ok_or(ArithmeticError::Overflow)?
-            .checked_add(&T::BaseDeposit::get())
-            .ok_or(ArithmeticError::Overflow)?;
-        Self::update_user_deposit(&user_account, payment_stream.user_deposit, new_deposit)?;
-
-        // Update the payment stream in the FixedRatePaymentStreams mapping
-        FixedRatePaymentStreams::<T>::mutate(provider_id, user_account, |payment_stream| {
-            let payment_stream = expect_or_err!(
-                payment_stream,
-                "Payment stream should exist if it was found before.",
-                Error::<T>::PaymentStreamNotFound
-            );
-            payment_stream.rate = new_rate;
-            payment_stream.user_deposit = new_deposit;
-            Ok::<(), DispatchError>(())
-        })?;
+        // The payment stream could have been deleted when charged if the user was out of funds.
+        // If that didn't happen, we update it with the new rate here.
+        if FixedRatePaymentStreams::<T>::get(provider_id, user_account).is_some() {
```

Comment:
Before, in this same function, we have the check:
```rust
        // Check that a fixed-rate payment stream between the Provider and User exists
        ensure!(
            FixedRatePaymentStreams::<T>::contains_key(provider_id, user_account),
            Error::<T>::PaymentStreamNotFound
        );
```

Doesn't that mean that if the user doesn't have a payment stream with the provider, the function will fail before even reaching this point? And if so, if an MSP is trying to remove a file from an insolvent user for a second time, he won't be able to do it, because the check will fail.

If I'm correct, please add a test case for this scenario. Also if this is the case, one thing we could do is handle this specific situation in the File System pallet, where we would check if the payment stream exists before reducing the bucket's size. If it doesn't, we handle it and accept it.